camayoc/api.py

```diff
@@ -0,0 +1,224 @@
+    def post(self, endpoint, payload, **kwargs):
+        """Send an HTTP POST request."""
+        url = urljoin(self.url, endpoint)
+        return self.request('POST', url, json=payload, **kwargs)
+
+    def put(self, endpoint, payload, **kwargs):
+        """Send an HTTP PUT request."""
+        url = urljoin(self.url, endpoint)
+        return self.request('PUT', url, json=payload, **kwargs)
+
+    def request(self, method, url, **kwargs):
+        """Send an HTTP request.
+
+        Arguments passed directly in to this method override (but do not
+        overwrite!) arguments specified in ``self.request_kwargs``.
+        """
+        # The `self.request_kwargs` dict should *always* have a "url" argument.
+        # This is enforced by `self.__init__`. This allows us to call the
+        # `requests.request` function and satisfy its signature:
+        #
+        #     request(method, url, **kwargs)
+        #
+        return self.response_handler(
+            self._cfg,
+            requests.request(method, url, **kwargs),
+        )
+
+
+class QCSClient(Client):
+    """An even easier client for interacting with the quipucords API.
+
+    This client inherits all methods and fields from api.Client and adds
+    more methods that protect the user from having to remember how to form
+    the proper relative paths to the different endpoints.
+
+    Example::
+        >>> from camayoc import api
+        >>> client = api.QCSClient()
+        >>> client.read_host_creds()
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Set base url and relative paths to API endpoints.
+
+        The call to super will set base url from the camayoc config file,
+        as well as set the response_handler that the client will use.
+        """
+        self.host_cred_path = 'credentials/hosts/'
+        self.net_prof_path = 'profiles/networks/'
+        super(QCSClient, self).__init__(*args, **kwargs)
+
+    def create_host_cred(self, payload):
+        """Send POST to QCS to create new host credential."""
+        return self.post(self.host_cred_path, payload)
+
+    def read_host_creds(self, host_cred_id=None):
+        """Send GET request to read host credentials.
+
+        If no host_cred_id is specified, get all host credentials,
+        otherwise just get the one specified.
+        """
+        path = self.host_cred_path
+        if host_cred_id:
+            path = urljoin(path, '{}/'.format(host_cred_id))
+        return self.get(path)
+
+    def update_host_cred(self, host_cred_id, payload):
+        """Send PUT request to update given host_cred_id with new data."""
+        path = urljoin(self.host_cred_path, '{}/'.format(host_cred_id))
+        return self.put(path, payload)
+
+    def delete_host_cred(self, host_cred_id):
+        """Send DELETE request for host_cred_id to QCS."""
+        path = urljoin(self.host_cred_path, '{}/'.format(host_cred_id))
+        return self.delete(path)
+
+    def create_net_prof(self, payload):
```

Comment:
Are profiles created for networks only? If so, perhaps we can name all methods by dropping the `net` part and fully spelling `profile`?